test/golden_tsickle_test.ts

```diff
@@ -28,9 +28,12 @@ const TEST_FILTER = (() => {
 })();
 
 // If true, update all the golden .js files to be whatever tsickle
-// produces from the .ts source. Do not change this code but run as:
-//     UPDATE_GOLDENS=y gulp test
-const UPDATE_GOLDENS = !!process.env.UPDATE_GOLDENS;
+// produces from the .ts source. Do not change this code but bazel run
+// the golden target instead of bazel test'ing it.
+// Detect that we are not running as a test, by using a well-known environment
+// variable. See https://docs.bazel.build/versions/master/test-encyclopedia.html
+
+const UPDATE_GOLDENS = !process.env.TEST_TMPDIR;
```

Comment:
I find this a bit spooky. Could we just have an explicit command line argument or environment variable passed in that controls this?

It's generally better to explicitly pass configuration into software than having software try to be clever and detect run modes from its local environment or other factors. In particular for something like this, where we could end up destroying data or end up with a test that never fails, I'd much rather be explicit.

Reply:
You already have to `bazel run` the target to get the goldens to update at all, because otherwise you're sandboxed and can't update the goldens.  If this magic fails you'll just get an error that you couldn't write the file, not a silently passing test.  We use this trick in g3 for our tsconfig golden tests, so I think we're on solid ground here.

Reply:
I don't like the pattern in google3 either. `bazel run` vs `bazel test` having the side effect of updating arbitrary test inputs is way too magic. An explicit flag here doesn't cost us anything and makes sure we and the user know what we're doing.

Reply:
Reverted, then.